app/models/target.rb

```diff
@@ -39,13 +40,30 @@ class Target < ActiveRecord::Base
                         numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
   validate :validate_target_limit, on: :create
 
+  reverse_geocoded_by :latitude, :longitude, address: :location
+
+  after_validation :reverse_geocode
+  after_create_commit :notify_compatible
+
   delegate :targets, to: :user, prefix: true
 
+  scope :near_targets, lambda { |target|
+    where(topic: target.topic)
+      .where.not(user_id: target.user_id)
+      .near([target.latitude, target.longitude], :area_lenght, radius: target.area_lenght)
+  }
+
   private
 
   def validate_target_limit
     return if user_targets.count < MAX_TARGETS_PER_USER
 
     errors.add(:targets, I18n.t('model.targets.errors.to_many'))
   end
+
+  def notify_compatible
+    users = Target.near_targets(self).map(&:user).uniq(&:id)
```

Comment:
Be careful, you could be mapping over a lot of records in memory. Take a look at this:
`User.where(id: Target.near_targets(self).pluck(:user_id))`